x/exchange/types/matching_test.go

```diff
@@ -23,21 +22,27 @@ func TestPayReceiveDenoms(t *testing.T) {
 
 func TestFillMemOrderBasic(t *testing.T) {
 	market := types.NewMarket(
-		1, "ucre", "uusd", utils.ParseDec("-0.0015"), utils.ParseDec("0.003"), utils.ParseDec("0.5"))
+		1, "ucre", "uusd",
+		utils.ParseDec("-0.0015"), utils.ParseDec("0.003"), utils.ParseDec("0.5"),
+		types.DefaultDefaultMinOrderQuantity, types.DefaultDefaultMinOrderQuote,
+		types.DefaultDefaultMaxOrderQuantity, types.DefaultDefaultMaxOrderQuote)
 	ctx := types.NewMatchingContext(market, false)
 
-	order := newUserMemOrder(1, true, utils.ParseDec("1.3"), sdk.NewDec(10_000000), sdk.NewDec(9_000000))
-	ctx.FillOrder(order, sdk.NewDec(5_000000), utils.ParseDec("1.25"), true)
-
-	require.True(t, order.IsMatched())
-	testutil.AssertEqual(t, sdk.NewDec(6_240625), order.Paid())
-	testutil.AssertEqual(t, sdk.NewDec(5_000000), order.Received())
-	testutil.AssertEqual(t, sdk.NewDec(-9375), order.Fee())
-
-	order = newUserMemOrder(2, false, utils.ParseDec("1.2"), sdk.NewDec(10_000000), sdk.NewDec(9_000000))
-	ctx.FillOrder(order, sdk.NewDec(5_000000), utils.ParseDec("1.25"), false)
-
-	testutil.AssertEqual(t, sdk.NewDec(5_000000), order.Paid())
-	testutil.AssertEqual(t, sdk.NewDec(6_231250), order.Received())
-	testutil.AssertEqual(t, sdk.NewDec(18750), order.Fee())
+	order := newUserMemOrder(1, true, utils.ParseDec("1.0015"), sdk.NewInt(10000), sdk.NewInt(9000))
+	ctx.FillOrder(order, sdk.NewInt(5500), utils.ParseDec("1.0013"), false)
+	res := order.Result()
+	utils.AssertEqual(t, sdk.NewInt(5500), res.ExecutedQuantity)
+	utils.AssertEqual(t, sdk.NewInt(5508), res.Paid)
+	utils.AssertEqual(t, sdk.NewInt(5483), res.Received)
+	utils.AssertEqual(t, sdk.NewInt(17), res.FeePaid)
+	utils.AssertEqual(t, sdk.NewInt(0), res.FeeReceived)
+
+	order = newUserMemOrder(2, false, utils.ParseDec("1.0015"), sdk.NewInt(10000), sdk.NewInt(9000))
+	ctx.FillOrder(order, sdk.NewInt(5500), utils.ParseDec("1.0013"), true)
```

Comment:
For sell order, the ordered price should be lower than the filled price.
But it seems that the ordered price (1.0015) is larger than the filled price (1.0013).

Reply:
You're right. Fixed the code.